src/main/java/project/config/JavaConfig.java

```diff
@@ -0,0 +1,37 @@
+package project.config;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+import javax.sql.DataSource;
+
+/**
+ * Created by slava23 on 10/19/2016.
+ */
+
+@Configuration
+@PropertySource(value = "application.properties")
+public class JavaConfig {
+
+    @Value("${spring.datasource.Driver}")
+    private String driverClassName;
+    @Value("${spring.datasource.url}")
+    private String url;
+    @Value("${spring.datasource.username}")
+    private String username;
+    @Value("${spring.datasource.password}")
+    private String password;
+
+    @Bean
+    public DataSource dataSource(){
```

Comment:
Let's add even more Spring Boot magic *** :)
Seems like Spring Boot can create datasource bean for you, see the first answer here:
http://stackoverflow.com/questions/28158094/spring-boot-enableautoconfiguration-with-exclude-not-working